src/Components/TodoList.tsx

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { Dispatch, RefObject } from 'react';
+import { Todo } from '../types/Todo';
+
+interface TodolistProps {
+  loadingTodoId: number[] | null;
+  setLoadingTodoId: Dispatch<React.SetStateAction<number[] | null>>;
+  loadingAllTodos: boolean;
+  setErrorMessage: Dispatch<React.SetStateAction<string>>;
+  setTodos: Dispatch<React.SetStateAction<Todo[]>>;
+  todos: Todo[];
+  inputRef: RefObject<HTMLInputElement>;
+  updateTodo: (updatedTodo: Todo) => Promise<void>;
+  editingTitle: string;
+  setEditingTitle: Dispatch<React.SetStateAction<string>>;
+  editingTodoId: number | null;
+  setEditingTodoId: Dispatch<React.SetStateAction<number | null>>;
```

Comment:
You can simplify your types by changing:
```suggestion
  setEditingTodoId: (value: number | null) => void
```